### DISCUSSION TITLE
Cropping a tiff file results in a file with the same size

### DISCUSSION THREAD

**Discussion by SuperRey** [ImageMagick/ImageMagick#6320] [Help]
### ImageMagick version

7.1.1-8 Q16-HDRI x64 920f792:20230422

### Operating system, version and so on

Windows

### Description

Hello,

I have a set of TIFF files that I want to split in two, I am using the following parameters

`magick IMG001.tif -crop 3100 +repage -rotate 180 +repage transformed\IMG001.1.tif`
`magick IMG001.tif -crop +3100 +repage -rotate 180 +repage transformed\IMG001.2.tif`

The cropping (and rotating) works well. However, the first resulting image (IMG001.1.tif) has the same file size as the original file. However the other one (IMG001.2.tif) has about half the size which is what I was expecting for both files.

Also, but I think it is not related, I get the following warning for the second command:

`magick: geometry does not contain image ("320x211+3100+0") 'IMG001.tif' @ warning/transform.c/CropImage/596.`

Am I missing something here? Anyway to get the expected file size?

In the following link you can find attached a link to the image:

https://www.dropbox.com/s/6owqqu8owb1iuxu/IMG001.tif?dl=0

Background, if it helps: These are film scans coming from a Nikon LS-4000 scanner. The camera I used uses a half-frame format, so for every frame scanned I get two pictures. I was trying to automate the split and rotation of the images.

**Comment by fmw42** [comment]
IM uses the default compression of none for the output. It does not remember the input compression. Check your input compression and use the same type compression for the output. That is add -compress XX where XX is what you find for the input.  The input may be compressed.

**Reply by SuperRey** [reply]
I already think about that, but both resulting images are about the same dimensions and the first one has the same size as the original, and the second one has about half the size.

The provided sample is just one of a set of 30 images (with different images on them so the complexity of the image should not be related) and all have the same issue.

**Comment by snibgo** [comment]
```
magick IMG001.TIF INFO:

IMG001.TIF[0] TIFF 5959x3946 5959x3946+0+0 16-bit Grayscale Gray 45.0442MiB 0.000u 0:00.000
IMG001.TIF[1] TIFF 320x211 320x211+0+0 8-bit TrueColor sRGB 0.000u 0:00.002
```
So the file contains two images. If you want only the first. use `IMG001.TIF[0]`.

`-crop 3100` doesn't do what you want. But I'm not sure what you do want.

**Reply by SuperRey** [reply]
Thanks, with your insights I now can avoid the warning message.

However the first split image is still having the same file size as the original file, even if it has around half the pixels.

Regarding the cropping I need, I will explain better what I am trying to do:

I have the original file like this:

![image](https://github.com/ImageMagick/ImageMagick/assets/4691049/ca98bdbb-dd7d-47c1-8c27-8bec09af5980)

As you can see, there are two subpictures. The vertical separation of the files is not always in the same place so it can vary, hence why I am not using 50%.

According to the crop and dimensions documentation, a single argument for the crop argument will be interpreted as width (exactly what I want) and the rest of dimensions will be computed accordingly (because I can rely on fixed width and height values). For the second command I am using +width. These part is working properly, as the output images have the expected dimensions.

Again, the issue here is that the image output from the first command has the same file size as the original image while it should be around half that value, as in the output file from the second command.

I am under the impression that the first command, even if cropped, is still saving the whole image and maybe masking the image to be shown cropped.

What do you think?

**Reply by SuperRey** [reply]
Thanks for the INFO command, now I can see there is a difference between the two output files:

```
C:\R00022\transformed>magick IMG001.1.tif INFO:
IMG001.1.tif[0] TIFF 3100x3946 3100x3946+0+0 16-bit Grayscale Gray 44.8514MiB 0.000u 0:00.024
IMG001.1.tif[1] TIFF 2859x3946 2859x3946+0+0 16-bit Grayscale Gray 0.000u 0:00.027

C:\transformed>magick IMG001.2.tif INFO:
IMG001.2.tif TIFF 2859x3946 2859x3946+0+0 16-bit Grayscale Gray 21.5188MiB 0.000u 0:00.001
```

It seems that for the first file the remaining of the crop is being saved in other subpicture, how can I avoid that?

**Reply by SuperRey** [reply]
I found the solution, thanks to this (https://stackoverflow.com/a/37572965/4278283) StackOverflow question:

```
magick IMG001.tif -crop 3100x+0+0 +repage -rotate 180 transformed\IMG001.1.tif
magick IMG001.tif -crop +3100+0 +repage -rotate 180 transformed\IMG001.2.tif
```

To avoid creating multiple pages we need to add the rest of the offset parameters.

Now onto learning how to prevent EXIF information :)

Thanks!
